### PR TITLE
Bugfix so multilabel confusion matrix can plot for 2 or more labels

### LINKED ISSUE #2857: `torchmetrics.classification.MultilabelConfusionMatrix` throws an exception when plotting the confusion matrix for two labels
## 🐛 Bug

When using torchmetrics multilabel confusion matrix an attribute error is thrown on line 275 of torchmetrics/utilities/plot.py when using two labels.

```
plot.py", line 275, in plot_confusion_matrix
ax.set_title(f"Label {fig_label[i]}", fontsize=15)
**AttributeError: 'numpy.ndarray' object has no attribute 'set_title'**
```
### To Reproduce

<details>
  <summary>Code sample</summary>

```python

from torchmetrics.classification import MultilabelConfusionMatrix
confusion_matrix = MultilabelConfusionMatrix(
    num_labels=2,
    normalize='true'
)
confusion_matrix.update(torch.ones(2, 2).int(), torch.ones(2, 2).int())
confusion_matrix.plot()
```

</details>

### Expected behavior

Expected a matplotlib fig and an ax containing the confusion matrix for two labels to be returned.

### Actual behaviour

When using greater than 2 labels the plot will be returned (an exception is raised for one label which is expected)

### Environment

- TorchMetrics: 1.6.0
- Python & PyTorch Version:  3.10.12 and '2.5.1+cu124'
- Any other relevant information such as OS: Ubuntu 22.04

### Suggested fix

change the 'and'
```python
ax = axs[i] if rows != 1 and cols != 1 else axs
```

to 'or'
```python
ax = axs[i] if rows != 1 or cols != 1 else axs
```

This will return axs[i] if either rows or cols is greater than 2 since this will return an array. If rows ad cols is 1 then we still get the axs

### PR DESCRIPTION
**Pull request by podgorki**
## What does this PR do?

Fixes #2857
Multilabel plotting failed for two labels and succeeded for greater than two. No test existed for this added one but it also requires matplotlib so that was added to the test requirements

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2858.org.readthedocs.build/en/2858/

### CODE REVIEW

File: src/torchmetrics/utilities/plot.py [hunk 1]
```diff
@@ -270,7 +270,7 @@ def plot_confusion_matrix(
     fig, axs = plt.subplots(nrows=rows, ncols=cols, constrained_layout=True) if ax is None else (ax.get_figure(), ax)
     axs = trim_axs(axs, nb)
     for i in range(nb):
-        ax = axs[i] if rows != 1 and cols != 1 else axs
+        ax = axs[i] if rows != 1 or cols != 1 else axs
```
Comment (Borda) [inline src/torchmetrics/utilities/plot.py:273]:
```suggestion
        ax = axs[i] if (rows != 1 or cols != 1) else axs
```

File: requirements/_tests.txt [hunk 2]
```diff
@@ -20,3 +20,5 @@ cloudpickle >1.3, <=3.1.0
 scikit-learn ==1.2.*; python_version < "3.9"
 scikit-learn ==1.5.*; python_version > "3.8"  # we do not use `> =` because of oldest replcement
 cachier ==3.1.2
+
+matplotlib==3.9.3
```
Comment (Borda) [inline requirements/_tests.txt:24]: why the pin here? we have this range:
https://github.com/Lightning-AI/torchmetrics/blob/f1d8f531fc5cb1120fe463930f7cf909ee9c4617/requirements/visual.txt#L4

**Review by SkafteNicki — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -27,6 +27,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- Fixed plotting of multilabel confusion matrix ([#2858](https://github.com/PyTorchLightning/metrics/pull/2858))
```
Comment (Borda) [inline CHANGELOG.md:30]:
```suggestion
- Fixed plotting of multilabel confusion matrix ([#2858](https://github.com/PyTorchLightning/metrics/pull/2858))


```

**Review by lantiga — APPROVED** [review]
Good catch

**Pull request merged**